### PR TITLE
Re-order Workflow tasks on every task execution wave

### PR DESCRIPTION
**Pull request by cjkindel**

### CODE REVIEW

File: griptape/structures/workflow.py [hunk 1]
```diff
@@ -92,11 +92,11 @@ def insert_task(
 
     def try_run(self, *args) -> Workflow:
         self._execution_args = args
-        ordered_tasks = self.order_tasks()
         exit_loop = False
 
         while not self.is_finished() and not exit_loop:
             futures_list = {}
+            ordered_tasks = self.order_tasks()
```
Comment (dylanholmes) [inline griptape/structures/workflow.py:99]: Just curious, what problem did you encounter?

Comment (collindutter) [reply griptape/structures/workflow.py:99]: Adding Tasks to the Workflow while it's running. It's less about re-ordering the Tasks on each wave and more about re-fetching them.

Comment (cjkindel) [reply griptape/structures/workflow.py:99]: This is actually @vachillo's finding, so I don't have the code. But I understand it was dynamic task/DAG changes during a workflow's execution that necessitates the task order being re-indexed. They may have more details

Comment (dylanholmes) [reply griptape/structures/workflow.py:99]: sounds exciting

**Review by collindutter — APPROVED** [review]

**Pull request merged**